[Ye Pei Lin] iP

peilinye opened the pull request:
## Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Duke frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,
1. download it from [here](https://github.com/peilinye/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 🤩✨

And it is **FREE**!

Features:

 - [x] Managing tasks
 - [x] Managing deadlines
 - [x] Managing events
 - [ ] Reminders (coming soon)

---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

peilinye committed:
Level-1

peilinye committed:
Level-2

peilinye committed:
Level-3

peilinye committed:
Level 4

peilinye committed:
restructure code level 4

peilinye committed:
A-TextUiTesting

peilinye committed:
Level-5

peilinye committed:
Level-6

peilinye committed:
clean up code

peilinye committed:
Level-7

peilinye committed:
Level-8

peilinye committed:
Merge branch 'branch-Level-7'
Level-7

peilinye committed:
resolve merge conflicts

peilinye committed:
change what is written to file

peilinye committed:
A-MoreOOP

peilinye committed:
A-Packages

peilinye committed:
delete old files

peilinye committed:
add .equals methods

peilinye committed:
add JUnit tests

peilinye committed:
add JavaDoc comments

peilinye committed:
follow coding standards

peilinye committed:
add find functionality

peilinye committed:
Merge branch 'branch-A-JavaDoc'

peilinye committed:
resolve merge conflict from merging A-CodingStandard

peilinye committed:
resolve merge conflict from merging A-Level-9

peilinye committed:
add missing javadocs

ong-hongkai commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,146 @@
+     *
+     * @return the task at the specific index in the list
+     */
+    public Task get(int i) {
+        return this.listOfTasks.get(i);
+    }
```
Should the method name be changed to something that better describes this method?

ong-hongkai commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,75 @@
+     *
+     * @return the string representation of the Deadline instance
+     */
+    @Override
+    public String toString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d MMM yyyy HH:mm");
+        return "[D]" + super.toString() + " (by: " + this.by.format(formatter) + ")";
+    }
```
I like the way you overrode the toString() method here.

ong-hongkai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+        }
+    }
+
+    public static void main(String[] args) {
+        new Duke().run();
```
I like the way you extracted the run method.

ong-hongkai reviewed (commented):
I went through your code and fortunately/unfortunately to you, I wasn't able to pick many coding violations at all. The only thing I felt could potentially be changed was the get() method found in TaskList. In fact, I managed to learn quite a bit just by reviewing your code, so great job on your IP!

peilinye committed:
add Gradle support

peilinye committed:
add checkstyle config files

peilinye committed:
fix some styling issues

cliveong commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,54 @@
+/**
+ * The class to represent a command to add tasks.
+ */
+public class AddCommand extends Command {
```
Perhaps use nouns instead of verbs as class names

cliveong commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,110 @@
+     * @return the list of tasks
+     * @throws DukeException
+     */
+    public ArrayList<Task> load() throws DukeException {
```
I think something along the lines of load existing tasks would give more details

cliveong reviewed (commented):
Great coding work. Perhaps am nitpick on the naming of the derived classes of Command, but the rest is great, have a lot to learn from!

peilinye committed:
add GUI

peilinye committed:
Merge branch 'branch-Level-10'

peilinye committed:
add welcome message

peilinye committed:
Add asertions

No assertions used to document important assumptions at various points of the code.

Important to add assertions to find potential bugs and terminate the program if an error occurs.

Let's
* use the assert feature at various points in the code

peilinye committed:
Refactor code

Code quality is not the best

Having good quality code will make it more understandable, and will make it easier for bug-searching.

Let's
* refactor the code in various ways

peilinye committed:
Merge pull request #1 from peilinye/branch-A-Assertions

Add assertions

peilinye committed:
Solve merge conflicts from merging master to branch-A-CodeQuality

peilinye committed:
Merge pull request #2 from peilinye/branch-A-CodeQuality

Refactor code to improve code quality

peilinye committed:
Set up Continuous Integration

peilinye committed:
Update build.gradle

peilinye committed:
Add support for undo command

peilinye committed:
remove import order checks from checkstyle

peilinye committed:
Add missing JavaDocs

peilinye committed:
Improve GUI

peilinye committed:
Add missing JavaDoc comments

peilinye committed:
Update JUnit test cases

peilinye committed:
Add user guide

peilinye committed:
Change exit command from bye to exit

peilinye committed:
Update build.gradle

peilinye committed:
Change where filepath is specified

peilinye committed:
Change JavaDoc header comments
